src/components/forms/ProxyDeviceInput.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useState } from "react";
+import { Input, Select } from "@canonical/react-components";
+
+interface Props {
+  onChange?: (type?: string, address?: string, port?: string) => void;
+  value?: string;
+  isConnect?: boolean;
+  isNatEnabled?: string;
+}
+
+const ProxyDeviceInput: FC<Props> = ({
+  onChange,
+  value,
+  isConnect,
+  isNatEnabled,
+}) => {
+  const [inheritedType, inheritedAddress, inheritedPort] =
+    value?.split(":") || [];
+
+  const [type, setType] = useState(inheritedType || "");
+  const [address, setAddress] = useState(inheritedAddress || "");
+  const [port, setPort] = useState(inheritedPort || "");
+
+  const AddressTypeOptions = [
+    {
+      label: "Select option",
+      value: "",
+      disabled: true,
+    },
+    {
+      label: "TCP",
+      value: "tcp",
+    },
+    {
+      label: "UDP",
+      value: "udp",
+    },
+    {
+      label: "UNIX",
+      value: "unix",
+    },
+  ];
+
+  return (
+    <>
+      <Select
+        label="Type"
+        onChange={(e) => {
+          const newType = e.target.value;
+          setType(newType);
+          onChange?.(newType, address, port);
```

Comment:
```suggestion
          onChange(newType, address, port);
```
We can directly update the parent state from here. Should work the same for the other fields below.